ibis/expr/datatypes/core.py

```diff
@@ -642,19 +642,11 @@ def to_integer_type(self):
 class Struct(DataType):
```

Comment:
We could also implement `Mapping` abc here, though I'm not sure whether that's desired.

Reply:
It would be nice to have `keys()` and `values()` instead of `names` and `types` or whatever it is that we implement now.